File: Source/OregonTilth.Models/DataTransferObjects/Workbook/Forms/CropYieldInformation/CropYieldInformationCreateDto.cs
using System.ComponentModel.DataAnnotations;

namespace OregonTilth.Models.DataTransferObjects
{
    public class CropYieldInformationCreateDto
    {
        [Required] public int WorkbookID { get; set; }
        [Required] public int CropID { get; set; }
        [Required] public int CropUnitID { get; set; }
        [Required] public decimal HarvestedYieldPerStandardUnitOfSpace { get; set; }
        [Required] public decimal MarketableYieldPerStandardUnitOfSpace { get; set; }
        [Required] public decimal PackagingCostPerCropUnit { get; set; }
        [Required] public decimal PricePerCropUnit { get; set; }
    }
}

File: OregonTilth.Web/src/app/shared/models/forms/crop-yield-information/crop-yield-information-create-dto.ts
export class CropYieldInformationCreateDto {
    WorkbookID: number;
    CropID: number;
    CropUnitID: number;
    HarvestedYieldPerStandardUnitOfSpace: number;
    MarketableYieldPerStandardUnitOfSpace: number;
    PackagingCostPerCropUnit: number;
    PricePerCropUnit: number;
    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}
